### PR TITLE
Add graph, model, tensor, and tensor_forward tests from C++

### PR DESCRIPTION
**Pull request by vbkaisetsu**
This branch partially adds test originally written in C++.

### CODE REVIEW

File: .travis/fedora.sh [hunk 1]
```diff
@@ -10,46 +10,59 @@ docker exec travis-ci bash -c "dnf update -y"
...
-docker exec travis-ci bash -c "git clone https://github.com/clMathLibraries/clBLAS.git"
-docker exec travis-ci bash -c "cd ./clBLAS/src && cmake . -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_TEST=OFF -DBUILD_KTEST=OFF"
-docker exec travis-ci bash -c "cd ./clBLAS/src && make && make install"
-# pocl 0.13 does not contain mem_fence() function that is used by primitiv.
-# We build the latest pocl instead of using distribution's package.
-# See: https://github.com/pocl/pocl/issues/294
-docker exec travis-ci bash -c "git clone https://github.com/pocl/pocl.git"
-docker exec travis-ci bash -c "cd ./pocl && cmake . -DCMAKE_INSTALL_PREFIX=/usr"
-docker exec travis-ci bash -c "cd ./pocl && make && make install"
+# OpenCL test is disabled in the following reasons:
```
Comment (odashi) [inline .travis/fedora.sh]: in the following reasons -> for the following reason

File: .travis/fedora.sh [hunk 2]
```diff
@@ -10,46 +10,59 @@ docker exec travis-ci bash -c "dnf update -y"
...
+#
+# 1. Irreproducible memory error is occured in Python+OpenCL+Fedora(+Travis?) combination.
+# 2. The test is run on the virtual machine. This situation is very rare.
```
Comment (odashi) [inline .travis/fedora.sh]: Remove this line, and add some documents what the developer should do to solve this problem.

**Review by odashi — COMMENTED** [review]
There are too many codes and I don't check the whole of them.

**Comment by vbkaisetsu** [comment]
Fixed.

File: .travis/fedora.sh [hunk 3]
```diff
@@ -10,10 +10,13 @@ docker exec travis-ci bash -c "dnf update -y"
 docker exec travis-ci bash -c "dnf install -y git rpm-build gcc-c++ cmake python3-devel python3-numpy"
 docker exec travis-ci bash -c "pip3 install cython scikit-build"
 
-# OpenCL test is disabled in the following reasons:
+# TODO:
```
Comment (odashi) [inline .travis/fedora.sh]: TODO(author)

**Comment by vbkaisetsu** [comment]
Fixed

**Review by odashi — APPROVED** [review]
LGTM

**Pull request merged**